src/geode/mesh/core/surface_mesh.cpp

```diff
@@ -249,6 +257,31 @@ namespace geode
             return absl::nullopt;
         }
 
+        void reset_polygons_around_vertex( index_t vertex_id )
+        {
+            polygons_around_vertex_->modify_value(
+                vertex_id, []( CachedPolygons& value ) { value.reset(); } );
+        }
+
+        const PolygonsAroundVertex& polygons_around_vertex(
+            const SurfaceMesh< dimension >& mesh,
+            index_t vertex_id,
+            absl::optional< PolygonVertex > first_polygon ) const

```

Comment:
Why not put the mesh as an attribute of the impl class ? In order for the mesh and impl class to be owners of their attributes ? ?